src/main/java/net/snowflake/ingest/utils/Constants.java

```diff
@@ -54,6 +54,7 @@ public class Constants {
   public static final String ENCRYPTION_ALGORITHM = "AES/CTR/NoPadding";
   public static final int ENCRYPTION_ALGORITHM_BLOCK_SIZE_BYTES = 16;
   public static final int MAX_STREAMING_INGEST_API_CHANNEL_RETRY = 3;
+  public static final int MAX_TOKEN_REFRESH_FOR_UNAUTHORIZED_RETRY = 3;
```

Comment:
I don't understand why we need a retry on this, if it works with one refresh then it will work, otherwise it will never work no matter how many times you retry, right?

Reply:
Make sense, removed the mutiple retries logic